### PR TITLE
Refactor test cases

### PR DESCRIPTION
**Pull request by rokshana-b**
**What changed?**
`wego app add --dry-run` & `wego app add` output changed

**Why?**
Because of these changes, some of the test cases had to be refactored

### CODE REVIEW

File: test/acceptance/test/utils.go [hunk 1]
```diff
@@ -330,12 +330,12 @@ func runWegoAddCommandWithOutput(repoAbsolutePath string, addCommand string, weg
 	return string(session.Wait().Out.Contents()), string(session.Wait().Err.Contents())
 }
 
-func runWegoAddCommandAndReturnSession(repoAbsolutePath string, addCommand string, wegoNamespace string) *gexec.Session {
-	command := exec.Command("sh", "-c", fmt.Sprintf("cd %s && %s %s", repoAbsolutePath, WEGO_BIN_PATH, addCommand))
-	session, err := gexec.Start(command, GinkgoWriter, GinkgoWriter)
-	Expect(err).ShouldNot(HaveOccurred())
-	return session
-}
+// func runWegoAddCommandAndReturnSession(repoAbsolutePath string, addCommand string, wegoNamespace string) *gexec.Session {
```
Comment (jpellizzari) [inline test/acceptance/test/utils.go:333]: Same here. Why is this commented out?

File: test/acceptance/test/add_tests.go [hunk 2]
```diff
@@ -147,6 +145,8 @@ var _ = Describe("Weave GitOps Add Tests", func() {
 		addCommand1 := "app add . --name=" + appName1
 		addCommand2 := "app add . --name=" + appName2
 
+		// var statusOutput string
```
Comment (jpellizzari) [inline test/acceptance/test/add_tests.go:148]: Leftover variable?

File: test/acceptance/test/add_tests.go [hunk 3]
```diff
@@ -337,14 +340,20 @@ var _ = Describe("Weave GitOps Add Tests", func() {
...
-			session = runWegoAddCommandAndReturnSession(repoAbsolutePath, addCommand, WEGO_DEFAULT_NAMESPACE)
+			addCommandOutput, _ = runWegoAddCommandWithOutput(repoAbsolutePath, addCommand, WEGO_DEFAULT_NAMESPACE)
 		})
 
 		By("Then I should see dry-run output with specified: url, namespace, branch", func() {
-			Eventually(session).Should(gbytes.Say("using URL: '" + url + "'"))
-			Eventually(session).Should(gbytes.Say("Checking cluster status... WeGOInstalled"))
-			Eventually(session).Should(gbytes.Say(`apiVersion:.*\nkind: GitRepository\nmetadata:\n\s*name: ` + appName + `\n\s*namespace: ` + WEGO_DEFAULT_NAMESPACE + `[a-z0-9:\n\s*]+branch: ` + branchName + `\n\s*.*\n\s*name: ` + appName + `\n\s*url: ` + url))
-			Eventually(session).Should(gbytes.Say(
+			Eventually(addCommandOutput).Should(MatchRegexp(`using URL: '` + url + `'`))
```
Comment (jpellizzari) [inline test/acceptance/test/add_tests.go:347]: Validating specific command-line output will make these tests very brittle. I realize that much of this suite already does that, so its not necessarily unique to this PR (nor should it block merging).

**Review by jpellizzari — COMMENTED** [review]
Looks like some left over commented-out code.

Side note: "trivial" changes to the output from the CLI will make these tests very noisy. I don't think anything needs to be done about it right now, but it speaks to a larger issue.

The concern is that noisy tests eat up cycles, get overlooked, or worse, turned off.

File: test/acceptance/test/utils.go (hunk 1 above)
Comment (rokshana-b) [reply test/acceptance/test/utils.go:333]: A useful function I could probably get rid of, for now

File: test/acceptance/test/add_tests.go [hunk 4]
```diff
@@ -214,6 +214,10 @@ var _ = Describe("Weave GitOps Add Tests", func() {
 			Expect(getRepoVisibility(os.Getenv("GITHUB_ORG"), appRepoName1)).Should(ContainSubstring("true"))
 			Expect(getRepoVisibility(os.Getenv("GITHUB_ORG"), appRepoName2)).Should(ContainSubstring("false"))
 		})
+
+		// By("When I run wego app status for running apps", func() {
```
[the remark this answers is not in the document]
Comment (rokshana-b) [reply test/acceptance/test/add_tests.go:218]: Yes, was trying to get the status of the apps, but `wego app status` is failing.

File: test/acceptance/test/add_tests.go (hunk 3 above)
Comment (rokshana-b) [reply test/acceptance/test/add_tests.go:347]: I realize that, but at the same time these outputs are user-facing scenarios. So, they need to be tested one way or another. These aren't part of the Smoke Tests, so they won't be blocking any merges. Any changes to these outputs may fail acceptance tests, but that just means we'll have to adjust the assertions being made.

File: test/acceptance/test/add_tests.go [hunk 5]
```diff
@@ -398,16 +401,17 @@ var _ = Describe("Weave GitOps Add Tests", func() {
...
+			verifyWorkloadIsDeployed("nginx", "my-nginx")
 		})
 
-		By("And I should see should see my workload deployed to the cluster", func() {
-			verifyWegoAddCommand(appName, WEGO_DEFAULT_NAMESPACE)
-			verifyWorkloadIsDeployed("nginx", "my-nginx")
+		By("And I verify my specified branch name", func() {
+			branchOutput := checkGitBranch(repoAbsolutePath)
+			Eventually(branchOutput).Should(ContainSubstring(branchName))
 		})
```
Comment (iahmad9) [inline test/acceptance/test/add_tests.go:415]: Is this verifying the local branch? Wondering if the test can verify the branch on the cluster?

Comment (rokshana-b) [reply test/acceptance/test/add_tests.go:415]: Not sure what you mean, but it's actually verifying the branch that I created and used during the `wego app add`. I wanted to make sure I added the app in my specific branch (test-branch).

**Review by iahmad9 — APPROVED** [review]
please make the requested change otherwise looks good.

**Pull request merged**